main.py

```diff
@@ -826,96 +831,282 @@ def on_link_down(self, event):
+        event_contents = {}
+        install_flows = {}
+        swapped_evcs = list[EVC]()
+        not_swapped_evcs = list[EVC]()
+
+        for evc in evcs:
+            install_flow = self.prepare_swap_to_failover_flow(evc)
+            if install_flow:
+                install_flows = merge_flow_dicts(install_flows, install_flow)
+                event_contents[evc.id] =\
+                    self.prepare_swap_to_failover_event(evc, install_flow)
+                swapped_evcs.append(evc)
+            else:
+                not_swapped_evcs.append(evc)
+
+        try:
+            send_flow_mods_event(
+                install_flows,
+                "install"
+            )
+            for evc in swapped_evcs:
+                temp_path = evc.current_path
+                evc.current_path = evc.failover_path
+                evc.failover_path = temp_path
+            emit_event(
+                self.controller, "failover_link_down",
+                content=deepcopy(event_contents)
+            )
+            return swapped_evcs, not_swapped_evcs
+        except FlowModException as exc:
+            log.error(f"Fail to install failover flows for {evcs}: {exc}")
+            return [], [*swapped_evcs, *not_swapped_evcs]
+
+    def prepare_clear_failover_flow(self, evc: EVC):
+        """Prepare an evc for clearing the old path."""
+        del_flows = {}
+        try:
+            del_flows = prepare_delete_flow(
+                merge_flow_dicts(
+                    evc._prepare_uni_flows(evc.failover_path, skip_in=True),
+                    evc._prepare_nni_flows(evc.failover_path)
+                )
+            )
+        # pylint: disable=broad-except
+        except Exception:
+            err = traceback.format_exc().replace("\n", ", ")
+            log.error(f"Fail to remove {evc} old_path: {err}")
+        return del_flows
+
+    def prepare_clear_failover_event(self, evc: EVC, delete_flow):
+        """Prepare event contents for clearing failover."""
+        return map_evc_event_content(
+            evc,
+            current_path=evc.current_path.as_dict(),
+            removed_flows=deepcopy(delete_flow)
+        )
+
+    def execute_clear_failover(
+        self,
+        evcs: list[EVC]
+    ) -> tuple[list[EVC], list[EVC]]:
+        """Process changes needed to commit clearing the failover path"""
+        event_contents = {}
+        delete_flows = {}
+        cleared_evcs = list[EVC]()
+        not_cleared_evcs = list[EVC]()
+
+        for evc in evcs:
+            delete_flow = self.prepare_clear_failover_flow(evc)
+            if delete_flow:
+                delete_flows = merge_flow_dicts(delete_flows, delete_flow)
+                event_contents[evc.id] =\
+                    self.prepare_clear_failover_event(evc, delete_flow)
+                cleared_evcs.append(evc)
+            else:
+                not_cleared_evcs.append(evc)
+
+        try:
+            send_flow_mods_event(
```

Comment:
The `send_flow_mods_http` needs to also be backported too, you probably are porting on flow_manager, but raising the point here again to be explicit and for us not to forget.